src/modules/persistence/users.persistence.service.ts

```diff
@@ -25,21 +25,21 @@ export class UsersPersistenceService extends Logger implements IUsersPersistence
     public async getAll(): Promise<UserDbModel[]> {
         try {
             this.log('UsersPersistenceService::getAll:: fetching users');
-            return this.collection.find({}).toArray();
+            return await this.collection.find({}).toArray();
         } catch (error) {
             this.error('UsersPersistenceService::getAll:: error fetching users', error.stack);
-            throw error;
+            throw  [error, null];
         }
     }
 
     public async getById(id: string): Promise<UserDbModel> {
         try {
             const _dbId = new ObjectID(id);
             this.log(`UsersPersistenceService::getAll:: fetching user by id ${id}`);
-            return this.collection.findOne({ _dbId });
+            return await this.collection.findOne({ _id: _dbId });
```

Comment:
if you will change the `_dbId` to `_id` this line can be simplified to ` return await this.collection.findOne({ _id });`

Reply:
fixed